inginious/frontend/plugins/upcoming_tasks/__init__.py

```diff
@@ -0,0 +1,148 @@
+import os
+from collections import OrderedDict
+from datetime import datetime, timedelta
+import flask
+
+
+from flask import send_from_directory
+from inginious.frontend.pages.utils import INGIniousPage, INGIniousAuthPage
+
+PATH_TO_PLUGIN = os.path.abspath(os.path.dirname(__file__))
+
+def menu(template_helper):
+    """ Displays the link to the board on the main page, if the plugin is activated """
+    return template_helper.render("main_menu.html", template_folder=PATH_TO_PLUGIN + '/templates/')
+
+class StaticMockPage(INGIniousPage):
+    """ MockPage based on auto-evaluation plugin structure """
+
+    def GET(self, path):
+        return send_from_directory(os.path.join(PATH_TO_PLUGIN, "static"), path)
+
+    def POST(self, path):
+        return self.GET(path)
+
+
+class UpComingTasksBoard(INGIniousAuthPage):
+
+    def GET_AUTH(self):
+        """ Called when reaching the page """
+        time_planner = "unlimited"
+        return self.page(time_planner)
+
+    def POST_AUTH(self):
+        """ Called when modifying time planner """
+        user_input = flask.request.form
+        time_planner = user_input.get("time_planner", default="unlimited")
+        return self.page(time_planner)
+
+    def time_planner_conversion(self, string_time_planner):
+        """ Used to convert the time_planner options into int value """
+        if string_time_planner in ["7", "14", "30"]:
+            return int(string_time_planner)
+        return 100000
+
+    def page(self, time_planner):
+        """ General main method called for GET and POST """
+        username = self.user_manager.session_username()
+        all_courses = self.course_factory.get_all_courses()
+        time_planner = self.time_planner_conversion(time_planner)
+
+        # Get the courses id
+        open_courses = {courseid: course for courseid, course in all_courses.items()
+                        if self.user_manager.course_is_open_to_user(course, username, False) and
+                        self.user_manager.course_is_user_registered(course, username)}
+        open_courses = OrderedDict(sorted(iter(open_courses.items()), key=lambda x: x[1].get_name(self.user_manager.session_language())))
+
+        # Get last submissions for left panel
+        last_submissions = self.submission_manager.get_user_last_submissions(5, {"courseid": {"$in": list(open_courses.keys())}})
+        except_free_last_submissions = []
+        for submission in last_submissions:
+            try:
+                submission["task"] = open_courses[submission['courseid']].get_task(submission['taskid'])
+                except_free_last_submissions.append(submission)
+            except:
+                pass
+
+        # Get the courses tasks, remove finished ones and courses that have no available unfinished tasks with upcoming deadline in range
+        tasks_data = {}
+        succeeded_courses = []
+        for courseid in open_courses:
```

Comment:
As this snippet is quite beefy, I'd rather have ``for courseid, course in open_courses.tems()`` too.

Reply:
Yep, nice little shortcut :)